lib/spack/spack/cmd/diff.py

```diff
@@ -89,6 +100,13 @@ def compare_specs(a, b, to_string=False, color=None):
         if func.name == "attr"
     )
 
+    if ignore_packages:
+        # If we are ignoring differences in dependencies, we probably want
+        # to also ignore differences in hashes (since any difference in a
+        # dep will transitively propagate to all parent hashes)
+        a_facts = set(x for x in a_facts if x.name not in ["hash"])
+        b_facts = set(x for x in b_facts if x.name not in ["hash"])
```

Comment:
I'm not convinced we should ignore the hashes here, but I don't know. If we add a spec-like syntax for this, that would naturally print in a way that hides the hash even if we leave it in.

Reply:
I felt it would be strange to show hashes since any change in a dependency would propagate up through all dependents, but I have removed this

> If we add a spec-like syntax for this, that would naturally print in a way that hides the hash even if we leave it in.

that seems like a reasonable way to handle it